backend/typescript/services/implementations/campService.ts

```diff
@@ -48,41 +38,76 @@ class CampService implements ICampService {
     }
   }
 
-  async createCamp(camp: CreateCampDTO, authId?: string): Promise<CampDTO> {
-    const baseCamp = new MgBaseCamp({
-      name: camp.name,
-      ageLower: camp.ageLower,
-      ageUpper: camp.ageUpper,
-      description: camp.description,
-      location: camp.location,
-      fee: camp.fee,
-      camperInfo: camp.camperInfo,
-    });
-    const newCamp = new MgCamp({
-      baseCamp,
-      campers: [],
-      capacity: camp.capacity,
-      waitlist: [],
-      startTime: camp.startTime,
-      endTime: camp.endTime,
-      dates: camp.dates,
-      active: camp.active,
-    });
-
+  async createCamp(camp: CreateCampDTO): Promise<CampDTO> {
+    let baseCamp: BaseCamp;
+    let newCamp: Camp;
+    const formQuestionIDs: string[] = [];
     try {
-      /* eslint no-underscore-dangle: 0 */
+      await Promise.all(
+        camp.formQuestions.map(async (formQuestion, i) => {
+          const question = await MgFormQuestion.create({
+            id: formQuestion.id,
+            type: formQuestion.type,
+            question: formQuestion.question,
+            required: formQuestion.required,
+            description: formQuestion.description,
+            options: formQuestion.options,
+          });
+          formQuestionIDs[i] = question._id;
```

Comment:
woah did not know you could just assign to whatever index in an array :o

Reply:
yeah LOL I had that same reaction